gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/task/KubernetesApplyTaskTest.java

```diff
@@ -53,8 +53,9 @@ public class KubernetesApplyTaskTest {
   @Before
   public void setUp() throws IOException {
     clusterAccessMockedConstruction = mockConstruction(ClusterAccess.class, (mock, ctx) -> {
-      final KubernetesClient kubernetesClient = mock(KubernetesClient.class);
+      final OpenShiftClient kubernetesClient = mock(OpenShiftClient.class);
```

Comment:
I am confused, why is it chaning to OpenShiftClient?

Reply:
This is required due to the change in the method signatures of the `KubernetesClient` DSL. The adapt method in L58 requires this mock to be an instance of `OpenShiftClient` and not `KubernetesClient`.

Most of the tests we have right have complex mocking setups which in part is why the tasks for replacing JMockit with Mockito are so complicated.

Reply:
ok no problem, but I think it deserves a comment in the code that says this is done in purpose and why. WDYT?

Reply:
Not against it, but I insist, this is not specific to this test but to almost all of the KubernetesClient related ones.

An intermediate, injectable layer in between the client and our specific usage would help.

Now that the Mock Server is also more reliable, some of the test could be migrated to use it with CRUD mode.